app/ui/shared/widgets/security-badge.jsx

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import omit from 'lodash/omit';
+
+import Btn from './btn';
+
+class SecurityBadge extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+  }
+
+  render() {
+
+    // Since we have unconfigurable security settings, if we access an HTTPS page and it loads,
+    // it is considered secure. Otherwise, all HTTP pages are insecure. Also, since
+    // we block all mixed content, this is even easier.
+    const image = /^https:/.test(this.props.url) ? 'ssl-secure.svg' : 'ssl-insecure.svg';
+
+    // If we have a tofino:// page, or the URL is empty (occurs during
+    // some page loads in our state), just hide the icon
+    let hidden = this.props.hidden;
+    if (/^tofino:/.test(this.props.url) || this.props.url === '') {
```

Comment:
Probably can take a Page and turn that into security info, but no need for that yet